libraries/ashrae/223p/nrel-templates/connections.yml

```diff
@@ -34,27 +34,19 @@ air-inlet-cp:
       s223:hasMedium s223:Medium-Air .
   optional: ["mapsto"]
 
-system-air-outlet-cp:
-  body: >
-    @prefix P: <urn:___param___#> .
-    @prefix s223: <http://data.ashrae.org/standard223#> .
-    P:name a s223:OutletSystemConnectionPoint ;
-      s223:hasMedium s223:Medium-Air .
-
-system-air-inlet-cp:
-  body: >
-    @prefix P: <urn:___param___#> .
-    @prefix s223: <http://data.ashrae.org/standard223#> .
-    P:name a s223:InletSystemConnectionPoint ;
-      s223:hasMedium s223:Medium-Air .
-
-
 duct:
   body: >
     @prefix P: <urn:___param___#> .
     @prefix s223: <http://data.ashrae.org/standard223#> .
     P:name a s223:Duct ;
-      s223:cnx P:a, P:b .
+      s223:hasMedium s223:Medium-Air ;
+      s223:connectsAt P:a, P:b .
+    # issue here is that 'connectsAt' requires a,b to be conn points
+    # but we can't instantiate that class directly *and* being a conn point
+    # involves other properties that must be included (e.g. hasmedium).
+    # TODO: how to solve this?
+    # P:a a s223:ConnectionPoint .
+    # P:b a s223:ConnectionPoint .
```

Comment:
Yeah, I feel like this kind of type hinting is inevitable. And it will also make suggestions much easier once we're at that point

Reply:
I wonder if a good way to do this would be with shacl conventions.

Reply:
I like this inherently, but I agree that it makes sense for another PR. I like that if we are explicitly using SHACL shapes we can use those shapes to search the existing graph for nodes which match the shape to suggest completion eventually.